sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/AsyncCacheNonBlocking.java

```diff
@@ -104,12 +116,20 @@ private Mono<TValue> getAsyncInternal(TKey key, Function<TValue, Mono<TValue>> s
                 if (initialLazyValue.shouldRemoveFromCache()) {
                     this.remove(key);
                 }
+
+                if (exception instanceof CosmosException) {
+                    CosmosException cosmosException = Utils.as(exception, CosmosException.class);
+                    logger.warn("initialLazyValue.getValueAsync [key exists] error for cache[{}] resulted in error with statusCode : [{}] and subsStatusCode : [{}]", key, cosmosException.getStatusCode(), cosmosException.getSubStatusCode());
+                } else {
+                    logger.warn("initialLazyValue.getValueAsync [key exists] error for cache[{}] resulted in error with message : [{}]", key, exception.getMessage());
+                }
+
                 logger.debug("cache[{}] resulted in error", key, exception);
```

Comment:
curious about these debug logs, I don't think we need to convert them to warn, but wanted to bring this to your attention.

Reply:
Yeah - these need not be converted - for the same `exception` instance, we do `logger.warn` (L121-L125).